Integrated dataloader to batch queries

Infirmarian opened the pull request:
Added loadtest `nearbyGQLQuery` to simulate querying for nearby artwork
Updated the api to utilize the dataloader
[Wiki](https://github.com/scalableinternetservices/wanderlust/wiki/Decoupling-and-Dataloader)

Infirmarian committed:
Integrated dataloader to batch queries
Added loadtest `nearbyGQLQuery` to simulate querying for nearby artwork
Updated the api to utilize the dataloader

Infirmarian committed:
Improved SQL nearby query performance by building an index on latitude and longitude columns
Added a python script to generate a LARGE number of randomly placed pieces of artwork
Added migration inserting a LARGE number of artwork and users

Infirmarian commented:
Whoops, I did the SQL stuff on the same branch. See the [Optimize SQL wiki](https://github.com/scalableinternetservices/wanderlust/wiki/Optimize-SQL)
closes #66

Timthetic commented on server/src/dataloaders/dataloader.ts:
```diff
@@ -0,0 +1,30 @@
+
+// const artLoader = new DataLoader()
+
+async function batchLoadUsers(keys: readonly number[]): Promise<(User | null)[]> {
+  const users = await User.find({ where: { id: In([...keys]) } })
+  const hashmap = new Map<number, User>()
+  users.forEach(u => hashmap.set(u.id, u))
+  return keys.map(k => hashmap.get(k) || null)
+}
+async function batchLoadArt(keys: readonly number[]): Promise<(Art | null)[]> {
+  const users = await Art.find({ where: { id: In([...keys]) } })
+  const hashmap = new Map<number, Art>()
+  users.forEach(u => hashmap.set(u.id, u))
+  return keys.map(k => hashmap.get(k) || null)
+}
+
+export interface Loaders {
+  user: DataLoader<number, User | null>
+  art: DataLoader<number, Art | null>
+}
+export function Loaders(): Loaders {
+  return {
+    user: new DataLoader<number, User | null>(keys => batchLoadUsers(keys)),
+    art: new DataLoader<number, Art | null>(keys => batchLoadArt(keys)),
+  }
+}
```
What a beaut.

Timthetic commented on server/src/db/generate_random_users.py:
```diff
@@ -0,0 +1,45 @@
+      for k in range(random.randint(args.minart, args.maxart + 1)):
+        r = args.radius * sqrt(random.random())
+        theta = random.random() * 2 * pi
+        art.append("('test_art', '{}', '{}', 'https://wanderlust-images.s3-us-west-2.amazonaws.com/f87d1363-cfad-4270-8f17-30c2fe671306', 1, {})".format(
+                    args.latitude + r * cos(theta),
+                    args.longitude + r * sin(theta),
+                    i))
```
I may be misinformed, but I think there is an unimportant mistake here that also exists in other portions of the app (that I wrote... womp). I have been informed that the earth is a "sphere" and that this does not "work the way I want it to". However, this should be pretty much a circle in most locations, so I'm not concerned 👍.

Also, I think this is the second kind of this Python script in the repo. If we cared about long term maintenance/cleanup, I would suggest you remove the other one since I like this one more.

Timthetic commented on server/src/graphql/api.ts:
```diff
@@ -45,27 +47,30 @@ export const graphqlRoot: Resolvers<Context> = {
-          lat: loc.lat,
-          lng: loc.lng,
-        })
+        .where(
+          'art.location.lat > :latLB AND art.location.lat < :latUB AND art.location.lng > :lngLB AND art.location.lng < :lngUB',
+          {
+            latLB: loc.lat - 0.02,
+            latUB: loc.lat + 0.02,
+            lngLB: loc.lng - 0.02,
+            lngUB: loc.lng + 0.02,
+          }
+        )
```
This was the other instance. As before, I think it's fine.

Timthetic commented on server/src/loadtest/nearbyGQLQuery.js:
```diff
@@ -0,0 +1,53 @@
+  let params = {
+    headers: {
+      'Content-Type': 'application/json',
+      'User-Agent': 'robert/k6',
```
Long term, this isn't something I would commit to the repository. Here, it's fine though.

Timthetic reviewed (approved):
Yeah, dataloader _does_ seem pretty cool. LGTM

Infirmarian committed:
Updated in response to comments on PR
Removed old python generation script
Setup k6 header as an environment variable for loadtests
